amazon-kinesis-client/src/main/java/software/amazon/kinesis/checkpoint/Checkpoint.java

```diff
@@ -26,18 +26,21 @@
 public class Checkpoint {
     private final ExtendedSequenceNumber checkpoint;
     private final ExtendedSequenceNumber pendingCheckpoint;
+    private final byte[] pendingCheckpointState;
```

Comment:
Why are we using raw byte array object to capture the state and not defining a concrete class instead ?

Reply:
Using a raw byte array since this will just be storing the application's serialized state; I didn't expect for us to be storing any other metadata along with that.